ArmRobot/Assets/Scripts/Robot/RobotVisionCapture.cs

```diff
@@ -0,0 +1,111 @@
+﻿using UnityEngine;
+using System.Collections;
+using System.Collections.Generic;
+
+public class RobotVisionCapture : MonoBehaviour
+{
+    public GameObject cube;
+    public GameObject cylinder;
+    public GameObject robot;
+   
+    public int nbMaxGameobjectScene = 10;
+    public float yAltitudeTable = 0.778f;
+    public float minimumDistanceWithoutObjects = 0.05f;
+    public float scaleObject = 0.1f;
+
+    
+    public float robotMinReach = 0.2f;
+    public float robotMaxReach = 0.5f;
+
+
+    void Start()
+    {
+        // first we set the position of the cube 
+        float minimumDistanceBetweenObjects = scaleObject + minimumDistanceWithoutObjects;
+
+        cube.transform.position = new Vector3(-robotMaxReach + minimumDistanceWithoutObjects, 
+                                        yAltitudeTable, -robotMaxReach + minimumDistanceWithoutObjects);
+        // then we set the position of the cylinder 
+        cylinder.transform.position = new Vector3(-robotMaxReach + minimumDistanceWithoutObjects + minimumDistanceBetweenObjects, 
+                                        yAltitudeTable, -robotMaxReach + minimumDistanceWithoutObjects + minimumDistanceBetweenObjects);
+        
+        // generate a random integer betwwen 1 and 10 but at least 1
+        int randomNumberGameObjects = 1 + Random.Range(1, nbMaxGameobjectScene); 
```

Comment:
The randomization of the number of distractor objects needs to happen after each datapoint is collected, not just in the Start() method (which is only called once).

Reply:
An efficient way to achieve this would be to create the max number of distractor objects in the beginning, and then enable/disable them during each iteration.